src/web/client/WebExtensionContext.ts

```diff
@@ -331,6 +350,14 @@ class WebExtensionContext implements IWebExtensionContext {
         }
 
         this._dataverseAccessToken = accessToken;
+
+        const websiteid = this.urlParametersMap.get(
+            Constants.queryParameters.WEBSITE_ID
+        ) as string;
+
+        const headers = getCommonHeaders(accessToken);
+
+        await this.populateSharedworkspace(headers, dataverseOrgUrl, websiteid);
```

Comment:
DataverseAuthentication() is not a good place to keep this code. It gets fired on all saves , individual file fetch, etc. 
authenticateAndUpdateDataverseProperties() will be the right place to call this. specially since it hooks onto the inital workspace create flow and will only bve called at initialization - which per my understanding is the intention here.

Reply:
Done: https://github.com/microsoft/powerplatform-vscode/pull/745/commits/474ee5dc652fef683911a8831052e808d9d44aa8